### PR TITLE
Add SMBIOS OEM strings support

### PR DESCRIPTION
**Pull request by liuw**
Fixes: #4319

### CODE REVIEW

File: arch/src/x86_64/smbios.rs [hunk 1]
```diff
@@ -134,11 +135,22 @@ pub struct SmbiosEndOfTable {
     pub handle: u16,
 }
 
+#[repr(C)]
+#[repr(packed)]
+#[derive(Default, Copy, Clone)]
+struct SmbiosOemString {
+    typ: u8,
```
Comment (rbradford) [inline arch/src/x86_64/smbios.rs:142]: I would call this `r#type`

**Review by rbradford — APPROVED** [review]
Could you also please try and remember `Fixes: #<..>` lines in commit messages or PR descriptions to auto link/close issues.

File: arch/src/x86_64/smbios.rs (hunk 1 above)
Comment (liuw) [reply arch/src/x86_64/smbios.rs:142]: Do you want to change other structures in the same file too? I can send a follow-up PR to fix them.

Re the Fixes tag, yes I can do that in the future.

Comment (rbradford) [reply arch/src/x86_64/smbios.rs]: Oops, sorry I didn't realise it was consistent within the same file. I would prefer it to use the `r#type` consistently. Feel free to push a commit to this PR.

Comment (liuw) [reply arch/src/x86_64/smbios.rs]: Sure no problem. I pushed two more clean-up patches.

File: tests/integration.rs [hunk 2]
```diff
@@ -3135,6 +3135,63 @@ mod parallel {
...
     }
 
+    #[test]
+    #[cfg(target_arch = "x86_64")]
+    fn test_dmi_oem_strings() {
+        let focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+        let guest = Guest::new(Box::new(focal));
+
+        let s1 = "io.systemd.credential:xx=yy";
+        let s2 = "This is a test string";
```
Comment (sboeuf) [inline tests/integration.rs:3145]: Ah cool, I'm glad this worked, I was worried the spaces inside the `s2` could have prevented the parsing from working correctly.

**Pull request merged**